tests/response/ExtractResponseFormatFromAcceptHeader.test.ts

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable max-len */
+import { extractResponseFormatFromAcceptHeader } from 'src'
+import { describe, expect, test } from 'vitest'
+
+describe('Test that response format is extracted correctly from accept header', () => {
+    test.each`
+        acceptHeader                                                   | returnJsonAsFallback | expectedResponseFormat
+        ${undefined}                                                   | ${true}              | ${'JSON'}
+        ${undefined}                                                   | ${false}             | ${'UNSUPPORTED'}
+        ${undefined}                                                   | ${undefined}         | ${'JSON'}
+        ${'somethingStrange'}                                          | ${true}              | ${'JSON'}
+        ${'somethingStrange'}                                          | ${false}             | ${'UNSUPPORTED'}
+        ${'application/graphql-response+json'}                         | ${true}              | ${'GRAPHQL-RESPONSE'}
+        ${'application/graphql-response+json'}                         | ${false}             | ${'GRAPHQL-RESPONSE'}
+        ${'application/graphql-response+json, application/json;q=0.9'} | ${true}              | ${'GRAPHQL-RESPONSE'}
+        ${'application/graphql-response+json, application/json;q=0.9'} | ${false}             | ${'GRAPHQL-RESPONSE'}
+        ${'application/json'}                                          | ${true}              | ${'JSON'}
+        ${'application/json'}                                          | ${false}             | ${'JSON'}
+        ${'application/json, application/graphql-response+json;q=0.9'} | ${true}              | ${'JSON'}
+        ${'application/json, application/graphql-response+json;q=0.9'} | ${false}             | ${'JSON'}
```

Comment:
Would a case where both variants have a q parameter be useful?

Reply:
We only support two content types in response and tried to keep the implementation as simple as possible without using an extra logic for the weight check. We assume that according to the Accept header specification no q parameter is interpreted as 1.